### PR TITLE
[hlo] algebraic simplifier select not pred pattern 

### PR DESCRIPTION
**Pull request by lipracer**
select(not(pred), on_true, on_false) => select(pred, on_false, on_true)

### CODE REVIEW

File: tensorflow/compiler/xla/service/algebraic_simplifier.cc
Comment (akuegel) [inline tensorflow/compiler/xla/service/algebraic_simplifier.cc:5368]: nit: typo, on_fasle -> on_false

**Review by akuegel — CHANGES_REQUESTED** [review]
Thanks for adding this simplification rule :)

**Review by akuegel — APPROVED** [review]

**Comment by lipracer** [comment]
I got below error message.
```
The specified bucket does not exist.
````

**Comment by akuegel** [comment]
> I got below error message.
> 
> ```
> The specified bucket does not exist.
> ```

I guess it is an infrastructure problem. I have notified our infrastructure team.

**Comment by lipracer** [comment]
> > I got below error message.
> > ```
> > The specified bucket does not exist.
> > ```
> 
> I guess it is an infrastructure problem. I have notified our infrastructure team.

Thanks.

**Comment by angerson** [comment]
There was indeed an error in our infrastructure configuration, which I've now fixed -- the "Details" link for that job was pointing to the wrong place. It should be fixed for any future check runs.

FYI, Code Check - Changed Files is not a blocking presubmit. It suggests clang and python format changes that may be needed to merge, but since we're still working on finding similar linter settings for internal and external presubmits it's not guaranteed that the suggested changes are necessary.

**Review by akuegel — COMMENTED** [review]
Now that we can see the presubmit error, it seems to indicate that algebraic_simplifier.cc has the wrong chmod config. It seems it has 755, but should have 644. Can you maybe check that, and if that is true, change it to 644?

**Comment by akuegel** [comment]
> Now that we can see the presubmit error, it seems to indicate that algebraic_simplifier.cc has the wrong chmod config. It seems it has 755, but should have 644. Can you maybe check that, and if that is true, change it to 644?

Ok, I just checked that this wrong mode is pre-existing, still it seems this blocks merging of this PR. I could also try to change it, but then you would have to rebase once my change lands (and I am not sure if rebasing will do the right thing). Please let me know if you need help with changing the mode.

**Comment by akuegel** [comment]
I have fixed the permissions now in f4b9d3dbb2d55b73a9a02b4e31cbbf9d02bee32e
Can you please rebase your PR? Lets hope this will keep the new file mode.

**Comment by lipracer** [comment]
Done.

**Review by akuegel — APPROVED** [review]

**Pull request merged**
